src/main/scala/io/qbeast/spark/index/query/QueryFiltersUtils.scala

```diff
@@ -0,0 +1,200 @@
+    condition.isInstanceOf[Or]
+  }
+
+  /**
+   * Recursively split Disjunctive operators (AND) in an expression
+   *
+   * @param condition the expression to evaluate
+   * @return
+   */
+
+  def splitDisjunctiveExpressions(condition: Expression): Seq[Expression] = {
+    condition match {
+      case Or(cond1, cond2) =>
+        splitDisjunctiveExpressions(cond1) ++ splitDisjunctiveExpressions(cond2)
+      case other => other :: Nil
+    }
+  }
+
+  /**
+   * Recursively split Conjunctive operators (OR) in an expression
+   *
+   * @param condition the expression to evaluate
+   * @return
+   */
+
+  def splitConjunctiveExpressions(condition: Expression): Seq[Expression] = {
+    condition match {
+      case And(cond1, cond2) =>
+        splitConjunctiveExpressions(cond1) ++ splitConjunctiveExpressions(cond2)
+      case other => other :: Nil
+    }
+  }
+
+  /**
+   * Convert an Spark String type to a Scala core type
+   * @param value the value to convert
+   * @return
+   */
+
+  def sparkTypeToCoreType(value: Any): Any = {
+    value match {
+      case s: UTF8String => s.toString
+      case _ => value
+    }
+  }
+
+  def isStringRangeExpression(expression: Expression): Boolean = {
+    val value = expression match {
+      case GreaterThan(_, l: Literal) => l
+      case GreaterThanOrEqual(_, l: Literal) => l
+      case LessThan(l: Literal, _) => l
+      case LessThanOrEqual(l: Literal, _) => l
+      case LessThan(_, l: Literal) => l
+      case LessThanOrEqual(_, l: Literal) => l
+      case GreaterThan(l: Literal, _) => l
+      case GreaterThanOrEqual(l: Literal, _) => l
+      case _ => return false
+    }
+
+    value.dataType.isInstanceOf[StringType]
+  }
+
+  /**
+   * Transform an expression
+   * Based on Delta DataSkippingReader
+   * Since we already know this filter are eligible for skipping
+   * we directly output the RangePredicate
+   *
+   * @param column
+   * @param values
+   * @return
+   */
+  def inToRangeExpressions(column: Expression, values: Seq[Any]): Seq[Expression] = {
+
+    val dataType = column.dataType
+
+    // If the type is String
+    // We need to apply first a Hash to understand what are the minimum and maximum
+    // that we need to search with Qbeast
+    if (dataType.isInstanceOf[StringType]) {
+      // Hash the values to match the search with Qbeast
+      val valuesHashed =
+        values.map(v => (v, MurmurHash3.bytesHash(v.asInstanceOf[UTF8String].getBytes)))
+
+      // Create ordering for new values
+      val ordering = new Ordering[(Any, Int)] {
+
+        /**
+         * Compare (Any, Int) by using Int comparison
+         */
+        override def compare(x: (Any, Int), y: (Any, Int)): Int =
+          Ordering[Int].compare(x._2, y._2)
+      }
+
+      val min = Literal(valuesHashed.min(ordering)._1, dataType)
+      val max = Literal(valuesHashed.max(ordering)._1, dataType)
+      Seq(LessThanOrEqual(column, max), GreaterThanOrEqual(column, min))
```

Comment:
Does it make sense? Hash(S1) < Hash(S2) does not imply S1 < S2 and vice versa. Let alone, that string comparison is a subtle matter, in many cases it depends on the locale.

Reply:
No, it does not, but in HashTransformation (string mapping), we hash the value of the string to determine the position in the space. 

What we are trying to evaluate here, is the range of space hashed that we need to retrieve with Qbeast. 

But today we discussed a bit and we found out that:
- Having a fixed size length for the strings helps to ensure that H(S1) < H(S2) imply S1 < S2. 
- Since we transform the value to a double, we must fit in 2^52. So the maximum number of characters to encode would be 11. (26^11 < 2^52)
- If the dataset has different values with length < 11, we will add 0s at the end of the string to fit the size. 
- If the dataset has different values with length > 11, they would be mapped in the same position, making the algorithm to read more data than the necessary. But at least, no records are lost.

Do you think our last approach makes sense?

Reply:
The approach you described above probably makes sense under the following assumptions:
1. All the strings are in Latin, if you have to include Cyrillic or Japanese symbols it will not work.
2. Even if you use Latin it is not completely clear how to compare strings, in general it depends on locale. Just a question which string is greater `esta` or `está`? The rule can depend on the national standards, i.e. locale. A more exotic examples is https://en.wikipedia.org/wiki/%C3%9F in German, as I remember it is a substitute for `ss` and in sorting should be treated as two letters, you may ask Clements for details.
At the same time I can imagine clients or which the string comparison you proposed is still useful, because it is better than nothing.

Reply:
Ideally the user should be aware of the constraints of their data. We can add an optional parameter to encode the strings with a certain length. 
So, by default the behaviour would be the current one: only a hash, with no possibility to execute range queries over that transformation. If the user finds the solution useful, she or he could configure it for each use case.

Reply:
Just one comment to start closing things:

Do you think we can merge this PR **with only the necessary changes** to include IN and OR predicate? So no touching anything related to Hash, and then open a new one with #188

Reply:
Feel free to merge anything you need to be merged. I have not started writing the code yet, so not problems.